cmd/age/age.go

```diff
@@ -174,19 +174,23 @@ func main() {
 		defer f.Close()
 		out = f
 	} else if terminal.IsTerminal(int(os.Stdout.Fd())) {
-		if armorFlag {
-			// If the output will go to a TTY, and it will be armored, buffer it
-			// up so it doesn't get in the way of typing the input.
+		if !armorFlag && name != "-" {
+			if decryptFlag {
+				// TODO: buffer the output and check it's printable.
+			} else {
+				// If the output wouldn't be armored, refuse to send binary to
+				// the terminal unless explicitly requested with "-o -".
+				logFatalf("Error: refusing to output binary to the terminal.\n" +
+					`Did you mean to use -a/--armor? Force with "-o -".`)
+			}
+		}
+		if terminal.IsTerminal(int(os.Stdin.Fd())) {
+			// If the input comes from a TTY and output will go to a TTY,
+			// and the output will be armored, buffer it up so it doesn't
+			// get in the way of typing the input.
```

Comment:
```
			// If the input comes from a TTY and output will go to a TTY,
			// buffer it up so it doesn't get in the way of typing the input.
```

Armoring is only half the story, and anyway it's not actually checked in this path, which is fine because who types binary into a TTY.

Reply:
Thanks for the hint, I somehow missed this. I'll update the pull request.